### DISCUSSION TITLE
Need some guidance on raspOVOS-gui.dev

### DISCUSSION THREAD

**Discussion by wita-sec** [OpenVoiceOS/ovos-buildroot#214] [Q&A]
Have installed OVOS on my RPi4 B from image
raspOVOS-gui.dev as of Feb 3rd 2024.

I have some basic problems with it.
Could be that I just do not know how it is supposed to work.
Or maybe there are some issues known with this build?

Here is what I have seen:

ovos-logs list -e shows some errors, e.g. something like:
Failed to load plugin entry point -
ovos_plugin_manager.utils:find_plugins:118
...
No module named 'ovos_utils.ocp'

ovos-gui did not start with the boot.
Also I was not asked for a wifi configuration.
Starting the gui manually returned an error:

(.venv) ovos@raspOvos:~ $ ovos-gui-app
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.

ovos-speak "hi michael" speaks to me, but
ovos-say-to "what is the ip" does not reply anything.

Should there be an ovos-skills service running? It appears there is no service like this on my machine.
I just used OVOS as it was flushed from the image.
Are there some further steps to follow unless skills are working? 

Some help would very much be appreciated to get me started with OVOS.

Thanks and regards,
Michael

**Comment by wita-sec** [comment]
Update:
Found out with 
  systemctl --user status ovos-skills
that skills have failed to start.
Run
  systemctl --user start ovos-skills
and was able to start skills.
Now ovos-say-to replies to me. 

One issue down.
How about the errors in the OVOS logs? 
How about starting the ovos gui?

**Reply by goldyfruit** [reply]
@builderjer

**Reply by builderjer** [reply]
The newest gui image is not working. You can try one of the older images at https://ovosimages.ziggyai.online/raspbian/development

**Comment by wita-sec** [comment]
Ok, thank you for your reply.
I'm just evaluating OVOS and plan to migrate my skills from Mycroft to OVOS.
Should I wait for a fix, provided it will come shortterm?
Otherwise I will switch to the latest build in development.

**Reply by builderjer** [reply]
I will try and get a working GUI image out this weekend.  That is one of the things on my TODO list.

**Comment by wita-sec** [comment]
BTW, how can I check the version of OVOS builds?

**Reply by builderjer** [reply]
I am not sure what you are getting at here, but this is what I think will answer this.

I have not yet started "versioning" the raspOVOS images as every build right now pulls in all of the latest packages from git.  There is a lot of behind the scenes work being done on OVOS versioning in general which will make giving these images a version much easier.

The Buildroot image is completely built from source, so it has pined package versions that it uses.  I am not sure if he has started with versioning his images either, but I know it is on the list.

I do believe that the ovos-installer is being versioned, but not sure what the format of that one is yet.

As for OVOS its self, you can try the command
`pip list | grep -i ovos` and it should give you a list of all the ovos packages installed with their respective version.

**Comment by wita-sec** [comment]
Thank you for your take on this. It was very helpful.
